components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,357 @@
+---@param numberOfGames integer
+---@return table
+function MatchFunctions.getExtraData(match, numberOfGames)
+	local extradata = {
+		t1bans = MatchFunctions.readBans(match.t1bans),
+		t2bans = MatchFunctions.readBans(match.t2bans),
+		mvp = MatchGroupInputUtil.readMvp(match),
+	}
+
+	local prefix = 'subgroup%d+'
+	Table.mergeInto(extradata, Table.filterByKey(match, function(key) return key:match(prefix .. 'header') end))
+	Table.mergeInto(extradata, Table.filterByKey(match, function(key) return key:match(prefix .. 'iskoth') end))
+
+	local t1Bans = Table.filterByKey(match, function(key) return key:match(prefix .. 't1bans') end)
+	Table.mergeInto(extradata, Table.mapValues(t1Bans, MatchFunctions.readBans))
+
+	local t2Bans = Table.filterByKey(match, function(key) return key:match(prefix .. 't2bans') end)
+	Table.mergeInto(extradata, Table.mapValues(t2Bans, MatchFunctions.readBans))
+
+	return extradata
+end
+
+---@param bansInput string
+---@return table
+function MatchFunctions.readBans(bansInput)
+	local bans = CustomMatchGroupInput._readCards(bansInput)
+
+	return Logic.nilIfEmpty(bans)
+end
+
+---@param mapInput table
+---@param mapIndex integer
+---@param subGroup integer
+---@param opponentCount integer
+---@return table
+---@return integer
+function MapFunctions.readMap(mapInput, mapIndex, subGroup, opponentCount)
+	subGroup = tonumber(mapInput.subgroup) or (subGroup + 1)
+
+	local map = {
+		map = 'Set ' .. mapIndex, -- CR has no map names, use generic one instead
+		subgroup = subGroup,
+	}
+
+	map.finished = MatchGroupInputUtil.mapIsFinished(mapInput)
+	local opponentInfo = Array.map(Array.range(1, opponentCount), function(opponentIndex)
+		local score, status = MatchGroupInputUtil.computeOpponentScore({
+			walkover = mapInput.walkover,
+			winner = mapInput.winner,
+			opponentIndex = opponentIndex,
+			score = mapInput['score' .. opponentIndex],
+		}, MapFunctions.calculateMapScore(mapInput, map.finished))
+		return {score = score, status = status}
+	end)
+
+	map.scores = Array.map(opponentInfo, Operator.property('score'))
+
+	if map.finished then
+		map.resulttype = MatchGroupInputUtil.getResultType(mapInput.winner, mapInput.finished, opponentInfo)
+		map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+		map.winner = MatchGroupInputUtil.getWinner(map.resulttype, mapInput.winner, opponentInfo)
+	end
+
+	return map, subGroup
+end
+
+---@param mapInput table
+---@param finished boolean
+---@return fun(opponentIndex: integer): integer?
+function MapFunctions.calculateMapScore(mapInput, finished)
+	local winner = tonumber(mapInput.winner)
+	return function(opponentIndex)
+		local manualScore = tonumber(mapInput['score' .. opponentIndex])
+		if manualScore then
+			return manualScore
+		end
+
+		-- TODO Better to check if map has started, rather than finished, for a more correct handling
+		if not winner and not finished then
+			return
+		end
+
+		return winner == opponentIndex and 1 or 0
+	end
+end
```

Comment:
Can map scores be > 1?

Reply:
no idea, i am not talking/writing with the troll and he seems to be the only one active on that wiki
i just went by the "old non git" version for this

Reply:
https://liquipedia.net/clashroyale/index.php?search=insource%3A%22score1%22
let's remove the manual score

Reply:
```suggestion
function MapFunctions.calculateMapScore(mapInput, finished)
	local winner = tonumber(mapInput.winner)
	return function(opponentIndex)
		-- TODO Better to check if map has started, rather than finished, for a more correct handling
		if not winner and not finished then
			return
		end

		return winner == opponentIndex and 1 or 0
	end
end
```